### PR TITLE
Allow setting properties on Datapoint objects

### PR DESCRIPTION
**Pull request by tedoc2000**

### CODE REVIEW

File: datapoint/datapoint.go [hunk 1]
```diff
@@ -88,6 +88,39 @@ func (dp *Datapoint) String() string {
...
+
+//SetProperty sets a property to be used when an MTS when it is created
+func (dp *Datapoint) SetProperty(key string, value string) {
+	if dp.Meta[stringProperties] == nil {
+		dp.Meta[stringProperties] = make(map[string]string, 1)
+	}
+	dp.GetProperties()[key] = value
+}
+
+//RemoveProperty remove a property from the map of properties which are used an MTS when it is created
```
Comment (cep21) [inline datapoint/datapoint.go]: Reenglish this please

File: datapoint/datapoint.go [hunk 2]
```diff
@@ -88,6 +88,39 @@ func (dp *Datapoint) String() string {
 	return fmt.Sprintf("DP[%s\t%s\t%s\t%d\t%s]", dp.Metric, dp.Dimensions, dp.Value, dp.MetricType, dp.Timestamp.String())
 }
 
+type metadata int
+
+const (
+	stringProperties metadata = iota
+)
+
+//SetProperty sets a property to be used when an MTS when it is created
```
Comment (cep21) [inline datapoint/datapoint.go]: Replace "an MTS" with "the time series associated with the datapoint" to make it more system agnostic.

**Comment by mpetazzoni** [comment]
@tedoc2000 @philatsf @cep21 I know and I understand why we need this for our `sf_isPrequantized` use case, but I think one of the reasons we didn't allow for at first still stands: the semantics of passing properties with datapoints are confusing, and wasteful.

Confusing, because only if the MTS doesn't exist at the time the datapoint is received will those properties be added to the newly created MTS. If it already exists, they are just ignored and don't update the MTS's metadata; and it's really difficult for an emitter to know if the MTS already exists or not.

Wasteful, because most emitters will send the properties with every single datapoint, but for N-1 of them this data will just be dropped on the floor, wasting potentially quite a bit of bandwidth (not to mention network and CPU time on our end, increasing latency).

I'd like to know if we plan on having a better story here, and if not how this will be documented and communicated.

**Comment by cep21** [comment]
@mpetazzoni Can you please describe a better way to accomplish what we want.  Maybe in an email thread or hipchat?

**Pull request merged**